keras_nlp/layers/transformer_decoder.py

```diff
@@ -223,28 +221,42 @@ def call(
         self_attended = self._add_and_norm(
             self_attended, decoder_sequence, self._decoder_attention_layernorm
         )
-        # Encoder-decoder attention.
-        encoder_decoder_attended = self._encoder_decoder_attention_layer(
-            query=self_attended,
-            value=encoder_sequence,
-            key=encoder_sequence,
-            attention_mask=encoder_mask,
-        )
-        encoder_decoder_attended = self._enc_dec_attentiondropout(
-            encoder_decoder_attended,
-        )
-        encoder_decoder_attended = self._add_and_norm(
-            encoder_decoder_attended,
-            self_attended,
-            self._enc_dec_attention_layernorm,
-        )
-        # Feedforward.
-        feed_forward_output = self._feed_forward(encoder_decoder_attended)
-        return self._add_and_norm(
-            encoder_decoder_attended,
-            feed_forward_output,
-            self._feedforward_layernorm,
-        )
+
+        if encoder_sequence is not None:
+            # Encoder-decoder attention.
+            encoder_mask = merge_padding_and_attention_mask(
+                encoder_sequence, encoder_padding_mask, encoder_attention_mask
+            )
+            encoder_decoder_attended = self._encoder_decoder_attention_layer(
+                query=self_attended,
+                value=encoder_sequence,
+                key=encoder_sequence,
+                attention_mask=encoder_mask,
+            )
+            encoder_decoder_attended = self._enc_dec_attentiondropout(
+                encoder_decoder_attended,
+            )
+            encoder_decoder_attended = self._add_and_norm(
+                encoder_decoder_attended,
+                self_attended,
+                self._enc_dec_attention_layernorm,
+            )
+            # Feedforward.
+            feed_forward_output = self._feed_forward(encoder_decoder_attended)
+            return self._add_and_norm(
+                encoder_decoder_attended,
+                feed_forward_output,
+                self._feedforward_layernorm,
+            )
+        else:
+            # Skip Encoder-Decoder attention, Feedforward.
```

Comment:
this is a bit confusing - the comma "," could suggest Feedforward is skipped as well.

Maybe just say "# Skip Encoder-Decoder attention if no `encoder_sequence` is provided."?